colossalai/inference/config.py

```diff
@@ -1,7 +1,73 @@
-"""
-Our config consists of three parts:
-    1. model_config: The configuration for the model, including `model name`, 'model path' and self-defined layer.
-    2. parallel_config: The configuration for parallelize model, including `tp_size`,'pp_size', `world size`, `local rank`, `master port`, `master ip`.
-    3. cache_config: Configuration for initialize and manage kv cache, including `block size`, `block num`
-For the convenience of users, we provide a unified config api for that wrapped all the configs. One can easily construct a colossal_config by setting the needed configs.
-"""
+from typing import Optional, Union
+
+import torch
+import torch.nn as nn
+
+
+class ColossalInferConfig:
+    """The infer configuration.
+
+    Args:
+    model: Path or nn.Module of this model.
+    tokenizer: Path of the tokenizer to use.
+    tokenizer_mode: "auto" will use the fast tokenizer if available, and "slow" will always use the slow tokenizer.
+    trust_remote_code: Whether to trust remote code from huggingface.
+    max_batch_size: Maximum batch size.
+    max_output_len: Maximum output length.
+    max_input_len: Maximum input length.
+    block_size: The number of blocks in a logical block.
+    gpu_utilization_rate: Maximum GPU memory usage ratio.
+    dtype: The data type for weights and activations.
+    tp_size: Tensor parallel size.
+    pp_size: Pipeline parallel size.
+    max_seq_len: Maximum length of input sentence.
+    quant_mode: Quantization mode.
+    revision: The specific version(a branch, name, a commit id, or a tag name) of model to use.
+    
+    
+    TODO To be modified as needed.
+    """
+
+    def __init__(
+        self,
+        model: Union[str, nn.Module],
+        tokenizer: str,
+        tokenizer_mode: str,
+        trust_remote_code: bool,
+        max_batch_size: int,
+        max_output_len: int,
+        max_input_len: int,
+        block_size: int,
+        gpu_utilization_rate: float,
+        dtype: Union[str, torch.dtype],
+        tp_size: int = 1,
+        pp_size: int = 1,
+        max_seq_len: Optional[int] = None,
+        quant_mode: Optional[str] = None,
+        revision: Optional[str] = None,
+    ):
+        self.model = model
+        self.tokenizer = tokenizer
+        self.tokenizer_mode = tokenizer_mode
+        self.trust_remote_code = trust_remote_code
+        self.max_batch_size = max_batch_size
+        self.max_output_len = max_output_len
+        self.max_input_len = max_input_len
+        self.block_size = block_size
+        self.gpu_utilization_rate = gpu_utilization_rate
+        self.tp_size = tp_size
+        self.pp_size = pp_size
+        self.dtype = dtype
+        self.max_seq_len = max_seq_len
+        self.quant_mode = quant_mode
+        self.revision = revision
```

Comment:
Such assignment is rather redundant, why not use `dataclass`?

Reply:
You can call verify args in `__post_init__` of `dataclass`

https://stackoverflow.com/questions/55183333/how-to-use-an-equivalent-to-post-init-method-with-normal-class

Reply:
Done